Add the support of CREATE FUNCTION

file `create-function.sql`:
```sql
CREATE FUNCTION linear_equation AS (x, k, b) -> k*x + b;
```

```bash
clickhouse-sql-parser -format -f  create-function.sql
```
output:
```bash
panic: parse statements error: line 0:7 <EOF> or ';' was expected, but got: "FUNCTION"
CREATE FUNCTION linear_equation AS (x, k, b) -> k*x + b;
       ^^^^^^^^
```
[UDF in ClickHouse documentation ](https://clickhouse.com/docs/en/sql-reference/statements/create/function)

--- a/parser/ast.go
+++ b/parser/ast.go
@@ -861,6 +861,41 @@ func (c *CreateView) String(level int) string {
 	return builder.String()
 }
 
+type CreateFunction struct {
+	CreatePos    Pos
+	FunctionName *Ident
+	OnCluster    *OnClusterExpr
+	Params       *ParamExprList
+	Expr         Expr
+}
+
+func (c *CreateFunction) Type() string {
+	return "FUNCTION"
+}
+
+func (c *CreateFunction) Pos() Pos {
+	return c.CreatePos
+}
+
+func (c *CreateFunction) End() Pos {
+	return c.Expr.End()
+}
+
+func (c *CreateFunction) String(level int) string {
+	var builder strings.Builder
+	builder.WriteString("CREATE FUNCTION ")
+	builder.WriteString(c.FunctionName.String(level))
+	if c.OnCluster != nil {
+		builder.WriteString(NewLine(level))
+		builder.WriteString(c.OnCluster.String(level))
+	}
+	builder.WriteString(" AS ")
+	builder.WriteString(c.Params.String(level))
+	builder.WriteString(" -> ")
+	builder.WriteString(c.Expr.String(level))
+	return builder.String()
+}
+
 type DestinationExpr struct {
 	ToPos           Pos
 	TableIdentifier *TableIdentifier

--- a/parser/parser_table.go
+++ b/parser/parser_table.go
@@ -15,6 +15,8 @@ func (p *Parser) parseDDL(pos Pos) (DDL, error) {
 		case p.matchKeyword(KeywordTable),
 			p.matchKeyword(KeywordTemporary):
 			return p.parseCreateTable(pos)
+		case p.matchKeyword(KeywordFunction):
+			return p.parseCreateFunction(pos)
 		case p.matchKeyword(KeywordMaterialized):
 			return p.parseCreateMaterializedView(pos)
 		case p.matchKeyword(KeywordLive):
@@ -1238,5 +1240,35 @@ func (p *Parser) parseTablePair(_ Pos) (*TablePair, error) {
 	return &TablePair{
 		Old: oldTable,
 		New: newTable,
+	}, nil
+}
+
+func (p *Parser) parseCreateFunction(pos Pos) (*CreateFunction, error) {
+	if err := p.consumeKeyword(KeywordFunction); err != nil {
+		return nil, err
+	}
+	functionName, err := p.parseIdent()
+	if err != nil {
+		return nil, err
+	}
+	if err := p.consumeKeyword(KeywordAs); err != nil {
+		return nil, err
+	}
+	params, err := p.parseFunctionParams(p.Pos())
+	if err != nil {
+		return nil, err
+	}
+	if _, err := p.consumeTokenKind(TokenArrow); err != nil {
+		return nil, err
+	}
+	expr, err := p.parseExpr(p.Pos())
+	if err != nil {
+		return nil, err
+	}
+	return &CreateFunction{
+		CreatePos:    pos,
+		FunctionName: functionName,
+		Params:       params,
+		Expr:         expr,
 	}, nil
 }

--- a/parser/parser_table.go
+++ b/parser/parser_table.go
@@ -1251,6 +1251,10 @@ func (p *Parser) parseCreateFunction(pos Pos) (*CreateFunction, error) {
 	if err != nil {
 		return nil, err
 	}
+	onCluster, err := p.tryParseOnCluster(p.Pos())
+	if err != nil {
+		return nil, err
+	}
 	if err := p.consumeKeyword(KeywordAs); err != nil {
 		return nil, err
 	}
@@ -1268,6 +1272,7 @@ func (p *Parser) parseCreateFunction(pos Pos) (*CreateFunction, error) {
 	return &CreateFunction{
 		CreatePos:    pos,
 		FunctionName: functionName,
+		OnCluster:    onCluster,
 		Params:       params,
 		Expr:         expr,
 	}, nil

--- a/parser/testdata/ddl/create_function_simple.sql
+++ b/parser/testdata/ddl/create_function_simple.sql
@@ -1,0 +1,1 @@
+CREATE FUNCTION linear_equation AS (x, k, b) -> k*x + b;

--- a/parser/testdata/ddl/format/create_function_simple.sql
+++ b/parser/testdata/ddl/format/create_function_simple.sql
@@ -1,0 +1,5 @@
+-- Origin SQL:
+CREATE FUNCTION linear_equation AS (x, k, b) -> k*x + b;
+
+-- Format SQL:
+CREATE FUNCTION linear_equation AS (x, k, b) -> k * x + b;

--- a/parser/testdata/ddl/output/create_function_simple.sql.golden.json
+++ b/parser/testdata/ddl/output/create_function_simple.sql.golden.json
@@ -1,0 +1,70 @@
+[
+  {
+    "CreatePos": 0,
+    "FunctionName": {
+      "Name": "linear_equation",
+      "Unquoted": false,
+      "NamePos": 16,
+      "NameEnd": 31
+    },
+    "OnCluster": null,
+    "Params": {
+      "LeftParenPos": 35,
+      "RightParenPos": 43,
+      "Items": {
+        "ListPos": 36,
+        "ListEnd": 43,
+        "HasDistinct": false,
+        "Items": [
+          {
+            "Name": "x",
+            "Unquoted": false,
+            "NamePos": 36,
+            "NameEnd": 37
+          },
+          {
+            "Name": "k",
+            "Unquoted": false,
+            "NamePos": 39,
+            "NameEnd": 40
+          },
+          {
+            "Name": "b",
+            "Unquoted": false,
+            "NamePos": 42,
+            "NameEnd": 43
+          }
+        ]
+      },
+      "ColumnArgList": null
+    },
+    "Expr": {
+      "LeftExpr": {
+        "LeftExpr": {
+          "Name": "k",
+          "Unquoted": false,
+          "NamePos": 48,
+          "NameEnd": 49
+        },
+        "Operation": "*",
+        "RightExpr": {
+          "Name": "x",
+          "Unquoted": false,
+          "NamePos": 50,
+          "NameEnd": 51
+        },
+        "HasGlobal": false,
+        "HasNot": false
+      },
+      "Operation": "+",
+      "RightExpr": {
+        "Name": "b",
+        "Unquoted": false,
+        "NamePos": 54,
+        "NameEnd": 55
+      },
+      "HasGlobal": false,
+      "HasNot": false
+    }
+  }
+]

